### PR TITLE
feat/daemon spawn

### PR DESCRIPTION
**Pull request by coryschwartz**
Adds additional endpoints for starting new daemons.

- [x]  API endpoints (may change)
- [x]  subprocess exec for the "local" region, for testing functionality
- [x] kubernetes client in progress

### CODE REVIEW

File: controller/spawn/local.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	daemoncmd, ok := s.cmds[daemonid]
+	if !ok {
+		return nil
+	}
+	return daemonFromCmd(daemoncmd, daemonid)
+}
+
+func (s *LocalSpawner) List(regionid string) []*Daemon {
+	daemons := make([]*Daemon, 0)
+	for daemonid, cmd := range s.cmds {
```
Comment (gammazero) [inline controller/spawn/local.go:80]: In other places, `s.Lock()` / `s.Unlock()` is used to protect access to `s.cmds`.  Should it be used here also?

File: controller/spawn/local.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		log.Info("dealbot exited normally", "id", d.Id, "code", cmd.ProcessState.ExitCode())
+		s.Lock()
+		delete(s.cmds, d.Id)
+		s.Unlock()
+	}()
+	return nil
+}
+
+func (s *LocalSpawner) Get(regionid string, daemonid string) *Daemon {
+	daemoncmd, ok := s.cmds[daemonid]
```
Comment (gammazero) [inline controller/spawn/local.go:71]: In other places, `s.Lock()` / `s.Unlock()` is used to protect access to `s.cmds`.  Should it be used here also?

**Comment by gammazero** [comment]
This PR has the best title I have seen

**Review by willscott — APPROVED** [review]

**Comment by willscott** [comment]
there's some dependency this adds that wants something not in the alpine docker build world.
maybe linux headers?

```
# github.com/zondax/hid
In file included from /go/pkg/mod/github.com/zondax/hid@v0.9.0/libusb/libusb/os/linux_usbfs.c:43,
                 from /go/pkg/mod/github.com/zondax/hid@v0.9.0/hid_enabled.go:26:
/go/pkg/mod/github.com/zondax/hid@v0.9.0/libusb/libusb/os/linux_usbfs.h:24:10: fatal error: linux/types.h: No such file or directory
   24 | #include <linux/types.h>
      |          ^~~~~~~~~~~~~~~
```

**Pull request merged**